index.js

```diff
@@ -8,10 +8,53 @@ var setSize = hasSet && setSizeDescriptor && typeof setSizeDescriptor.get === 'f
 var setForEach = hasSet && Set.prototype.forEach;
 var booleanValueOf = Boolean.prototype.valueOf;
 
+function listFmtDefault(objType, size, parts, details) {
+  var parens = '{}', padType = ' ', padParts = ' ';
+  // size = null: The concept of size doesn't easily apply for this type.
+  // size = false: parts.length could be considered as the size but it's
+  //        customary to not display it for this type.
+
+  switch (objType || null) {
+  // We could save lots of cases and achieve more consistency in formating,
+  // but that would be a breaking change so let's defer that for the next
+  // breaking semver jump.
+  case null:
+    break;
+  case 'Array':
+    parens = '[]';
+    padType = objType = '';
+    break;
+  case 'Object':
+    padType = objType = '';
+    break;
+  case 'Error':
+    objType = '[' + details.obj + ']';  // NB: implicit String() conversion
+    if (!parts.length) {
+      parens = '';
+      padType = '';
+    }
+    break;
+  case 'Map':
+  case 'Set':
+    objType += ' (' + size + ') {';
+    size = null;
+    parens = [ '', '}' ];
+    padType = padParts = '';
+    break;
+  }
+  if (typeof size === 'number') { objType += ' (' + size + ')'; }
+  parts = parts.join(', ');
+  if (!parts) { padParts = ''; }
+  return (parens[0] || '') + padType + objType +
+    (padParts || padType) + parts + padParts +
+    (parens[1] || '');
+}
+
 module.exports = function inspect_ (obj, opts, depth, seen) {
     if (!opts) opts = {};
     
     var maxDepth = opts.depth === undefined ? 5 : opts.depth;
+    var listFmt = (opts.listFmt || listFmtDefault);
```

Comment:
I think this option name might make more sense as "listFormatter" or "formatList"?